cupy/cuda/function.pyx

```diff
@@ -155,7 +155,8 @@ cdef _launch(intptr_t func, Py_ssize_t grid0, int grid1, int grid2,
 
 cdef class Function:
 
-    """CUDA kernel function."""
+    """Class in charge of the execution of a compiled CUDA kernel function.
+    """
```

Comment:
Do you have to change the summary line?
Execution is not the only function of this class.
As we're adding kernel attributes, you may as well use this class to simply retrieve those attributes.

I think the previous expression represents exactly what this class is.